server/test/users.spec.js

```diff
@@ -0,0 +1,223 @@
+          token = res.body.token;
+          userId = res.body.userId;
+        });
+      server.post('/users')
+      .send(testData.adminUser)
+        .expect('Content-Type', /json/)
+        .expect(201)
+        .end((err, res) => {
+          should(res.body).have.property('token');
+          adminToken = res.body.token;
+          adminId = res.body.userId;
+          done();
+        });
+    });
+
+    it('ensures unique user is created each time', (done) => {
+      server.post('/users')
+      .send(testData.regularUser)
+        .expect('Content-Type', /json/)
+        .expect(400)
+        .end((err, res) => {
+          res.status.should.equal(400);
+          done();
+        });
+    });
+
+    it('should fail when all parameters are not given', (done) => {
+      server.post('/users')
+      .send(testData.badUser)
+        .expect('Content-Type', /json/)
+        .expect(400)
+        .end((err, res) => {
+          res.status.should.equal(400);
+          should(res.body).have.property('error');
+          done();
+        });
+    });
+  });
+
+  describe('on login', () => {
+    it('should give token to valid users', (done) => {
+      server.post('/login')
+      .send(testData.regularUser)
+        .expect('Content-Type', /json/)
+        .expect(200)
+        .end((err, res) => {
+          should(res.body).have.property('token');
+          done();
+        });
+    });
+
+    it('should throw an error for invalid users', (done) => {
+      server.post('/login')
+      .send(testData.regularUser2)
+        .expect('Content-Type', /json/)
+        .expect(400)
+        .end((err, res) => {
+          res.status.should.equal(400);
+          should(res.body).have.property('error');
+          done();
+        });
+    });
+  });
+
+  describe('edit user ', () => {
+    it('updates the user attributes', (done) => {
+      const newAttributes = {
+        firstName: 'John', lastName: 'Doe', email: 'johndoe@mail.com'
+      };
+      server.put(`/users/${userId}`)
+        .set({ 'x-access-token': token })
+        .send(newAttributes)
+        .end((err, res) => {
+          res.status.should.equal(200);
+          should(res.body.found.lastName).be.exactly(newAttributes.lastName);
+          done();
+        });
+    });
+
+    it('should return NOT FOUND for an invalid id', (done) => {
+      server.put('/users/100')
+        .set({ 'x-access-token': token })
+        .expect(404)
+        .end((err, res) => {
+          res.status.should.equal(404);
+          done();
+        });
+    });
+
+    it(`should send an error message 
+    if user is trying to update another user`, (done) => {
+      server.put(`/users/${adminId}`)
+        .set({ 'x-access-token': token })
+        .expect(401)
+        .end((err, res) => {
+          res.status.should.equal(401);
+          should(res.body).have.property('error');
+          done();
+        });
+    });
+
+    it('should return bad request for invalid access', (done) => {
```

Comment:
`Unauthorized` should be the word here since no `token` was provided with the request. As a side note, `Bad Request` should be used when a user missed a `notNull` attribute or maybe an error in how the request was structured and not necessarily an `authorization` issue. See [here](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#4xx_Client_Error) for more information